pennylane_lightning/src/tests/Test_KernelMap.cpp

```diff
@@ -137,3 +137,67 @@ TEST_CASE("Test KernelMap functionalities", "[KernelMap]") {
             Util::LightningException, "does not exist");
     }
 }
+
+TEST_CASE("Test KernelMap is consistent in extreme usecase", "[KernelMap]") {
+    using Gates::GateOperation;
+    using Gates::KernelType;
+    using EnumKernelMap =
+        OperationKernelMap<Gates::GateOperation>::EnumKernelMap;
+    auto &instance = OperationKernelMap<Gates::GateOperation>::getInstance();
+
+    const auto num_qubits = std::vector<size_t>{4, 6, 8, 10, 12, 14, 16};
```

Comment:
Should parameterise the tests based on num_qubits or is it fine to handle it in the same case?

Reply:
I don't think we need to do so. I just randomly generated an extreme case that calls `getKernelMap` lots of times simultaneously from multiple threads. This test case already raises an error for the previous code.